Automattic/jetpack
# Search: use centralized WPCOM search pricing API

Fixes #24448 #24686

#### Changes proposed in this Pull Request:
Refactors My Jetpack and Jetpack Search package to use WPCOM search pricing API ( D82514-code ) to prepare for the upcoming pricing change of search products, also to centralize logic, so that we'll be able to adjust the pricing from the WPCOM side, and the plugins would receive the new pricing straightly. And the PR also removes pricing estimation process in the search package which is not used anymore. 

#### Jetpack product discussion
pcNPJE-Tg-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Go to a site not connected
* Ensure search pricing is correct on page  `/wp-admin/admin.php?page=jetpack-search`
* Ensure search pricing is correct in My Jetpack `/wp-admin/admin.php?page=my-jetpack#/add-search`
